build-logic/src/main/kotlin/kotlin-inject.jvm.gradle.kts

```diff
@@ -1,13 +1,19 @@
 import org.gradle.accessors.dm.LibrariesForLibs
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 
 plugins {
     kotlin("jvm")
 }
 
 val libs = the<LibrariesForLibs>()
 
+kotlin {
+    jvmToolchain(17)
```

Comment:
I don't quite understand the distinction between the jvmToolchain and the target version. Is there a reason we need both and that they should be different?

Reply:
The toolchain version specifies which version of Java should be used to compile the project. The target version specifies the least compatible version of bytecode that will be produced (i.e. projects using at least Java 8 can use this library).